alpaca/broker/client.py

```diff
@@ -414,3 +483,107 @@ def _parse_activity(data: dict) -> Union[TradeActivity, NonTradeActivity]:
             return parse_obj_as(TradeActivity, data)
         else:
             return parse_obj_as(NonTradeActivity, data)
+
+    # ############################## TRADE ACCOUNT DOCUMENTS ################################# #
+
+    def get_trade_documents_for_account(
+        self,
+        account_id: Union[UUID, str],
+        documents_filter: Optional[GetTradeDocumentsRequest] = None,
+    ) -> List[TradeDocument]:
+        """
+        Gets the list of TradeDocuments for an Account.
+
+        Args:
+            account_id (Union[UUID, str]): The id of the Account you wish to retrieve documents for. str values will
+              attempt to be upcast into UUID instances
+            documents_filter (GetTradeDocumentsRequest): The optional set of filters you can apply to filter the
+              returned list.
+
+        Returns:
+            List[TradeDocument]: The filtered list of TradeDocuments
+        """
+        account_id = validate_uuid_id_param(account_id)
+
+        result = self.get(
+            f"/accounts/{account_id}/documents",
+            documents_filter.to_request_fields() if documents_filter else {},
+        )
+
+        return parse_obj_as(List[TradeDocument], result)
+
+    def get_trade_document_for_account_by_id(
+        self,
+        account_id: Union[UUID, str],
+        document_id: Union[UUID, str],
+    ) -> TradeDocument:
+        """
+        Gets a single TradeDocument by its id
+
+        Args:
+            account_id (Union[UUID, str]): The id of the Account that owns the document
+            document_id (Union[UUID, str]): The id of the TradeDocument
+
+        Returns:
+            TradeDocument: The requested TradeDocument
+        Raises:
+            APIError: Will raise an APIError if the account_id or a matching document_id for the account are not found.
+        """
+
+        account_id = validate_uuid_id_param(account_id)
+        document_id = validate_uuid_id_param(document_id, "document_id")
+
+        response = self.get(f"/accounts/{account_id}/documents/{document_id}")
+
+        return parse_obj_as(TradeDocument, response)
+
+    def download_trade_document_for_account_by_id(
+        self,
+        account_id: Union[UUID, str],
+        document_id: Union[UUID, str],
+        file_name: str,
+    ) -> None:
+
+        account_id = validate_uuid_id_param(account_id)
+        document_id = validate_uuid_id_param(document_id, "document_id")
+        response: Optional[Response] = None
+
+        # self.get/post/etc all set follow redirects to false, however API will return a 301 redirect we need to follow,
+        # so we just do a raw request
+
+        target_url = f"{self._base_url}/{self._api_version}/accounts/{account_id}/documents/{document_id}/download"
+        num_tries = 0
+
+        while num_tries <= self._retry:
```

Comment:
Why are we making the request here instead of using one of the methods within the base client? 

Is it because the base client returns it as a `.json` and we want to handle the raw response?

Also the base implementation adds a 3 second delay between repeated requests. At first I thought it was to handle any rate-limiting that might be in place. But Alpaca has a 200 requests per minute limit, and Broker specifically has a 1000 requests per minute limit. The delay in the base implementation seems redundant.

Reply:
Yeah this function does its own request call from scratch because of both the options the base class sets of not following redirects as well as forcing a .json out the end when what we want out of this is a raw stream to iterate over.

and yeah the 3 second delay isn't actually redundant and its not for the request limit. Well behaved clients should use progressive fallback for error codes to attempt to give the server time to recover. Realistically we should change the base implementation to wait progressively longer periods of time the more errors it gets but that can be a future fix

Reply:
Sounds good. I'll add a ticket to make the base class more flexible, so we can eventually move this request call to the base client.